File: src/gestorAplicacion/gestion/Facultad.java
package gestorAplicacion.gestion;

import java.io.Serializable;
import java.util.*;

import java.util.*;
import gestorAplicacion.gestion.*;
import gestorAplicacion.personas.*;

public class Facultad implements Serializable{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private String nombre;
	private ArrayList<Profesor> profesoresVinculados;
	private ArrayList<Estudiante> estudiantes;
	private static ArrayList<Facultad> listaFacultades = new ArrayList<Facultad>();

	public Facultad(String nombre, ArrayList<Profesor> profesoresVinculados, ArrayList<Estudiante> estudiantes) {
		this.nombre = nombre;
		this.setProfesoresVinculados(profesoresVinculados);
		this.setEstudiantes(estudiantes);
		listaFacultades.add(this);
	}


	public void vincularProfesor(Profesor profesor) {
		profesoresVinculados.add(profesor);
	}

	public void crearAsignatura(Asignatura asignatura) {

	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public ArrayList<Profesor> getProfesoresVinculados() {
		return profesoresVinculados;
	}

	public void setProfesoresVinculados(ArrayList<Profesor> profesoresVinculados) {
		this.profesoresVinculados = profesoresVinculados;
		for (Profesor p: profesoresVinculados) {p.setFacultad(this);}
	}

	public ArrayList<Estudiante> getEstudiantes() {
		return estudiantes;
	}

	public void setEstudiantes(ArrayList<Estudiante> estudiantes) {
		for (Estudiante e: estudiantes) {e.setFacultad(this);}

		this.estudiantes = estudiantes;
	}

	public static ArrayList<Facultad> getListaFacultades() {
		return listaFacultades;
	}

	public static void setListaFacultades(ArrayList<Facultad> listaFacultades) {
		Facultad.listaFacultades = listaFacultades;
	}
}

File: srcpy/gestorAplicacion/gestion/facultad.py
class Facultad:
    listaFacultades = []
    profesoresVinculados = []

    def __init__(self, nombre, profesoresVinculados, estudiantes):
        from gestorAplicacion.personas.estudiante import Estudiante
        from gestorAplicacion.personas.profesor import Profesor

        self.nombre = nombre
        self.setProfesoresVinculados(profesoresVinculados)
        self.setEstudiantes(estudiantes)
        Facultad.listaFacultades.append(self)

    def vincularProfesor(self, profesor):
        self.profesoresVinculados.append(profesor)

    def crearAsignatura(self, asignatura):
        pass

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def getProfesoresVinculados(self):
        return self.profesoresVinculados

    def setProfesoresVinculados(self, profesores):
        from gestorAplicacion.personas.profesor import Profesor

        self.profesoresVinculados = profesores
        for profesor in self.profesoresVinculados:
            profesor.setFacultad(self)

    def getEstudiantes(self):
        return self.estudiantes

    def setEstudiantes(self, estudiantes):
        for estudiante in estudiantes:
            estudiante.setFacultad(self)
        self.estudiantes = estudiantes

    @classmethod
    def getListaFacultades(cls):
        return cls.listaFacultades

    @classmethod
    def setListaFacultades(cls, facultades):
        cls.listaFacultades = facultades
